### PR TITLE
MRG, ENH: Add some troubleshooting to install docs

### LINKED ISSUE #6365: DOC: Installation troubleshooting
Talking to @jasmainak about an upcoming workshop and getting emails, I think we would benefit from a dedicated page about troubleshooting installation issues. A page like this could be nice:

http://ryan-leung.github.io/PHYS4650_Python_Tutorial/installing-on-linux.html

Basically something with green/checkboxes saying "this is good" and red/X's saying "this is bad" at different stages, with links to "how to fix bad" would be helpful. I'm not sure how it should be structured. @drammock do you want to try, or would you rather I take a stab at it?

Maybe this should be broken into:

1. Ensure conda is installed and accessible from your terminal
2. Ensure the `mne` environment is active
3. Ensure the dependencies are installed (`sys_info`, maybe `mlab.test_plot3d`)
4. Ensure the dependencies work (esp. `mayavi`, with interactivity / OpenGL issues)
5. Ensure Spyder works (?) if we want to push people that way

Example diagnostics that should appear somewhere, with what they imply:

- `which conda` (good: correct path, bad: no output)
- `conda 
- `which python` (good: Anaconda, bad: anything else e.g. system python)
- `which pip` (good: Anaconda pip, bad: anything else e.g. system pip)
- `echo $PATH` (good: Anaconda on path; bad: not on path)
- `python -c "import mne"` (good: works, bad: fails)
- `echo $PYTHONPATH` (good: empty, bad: could point somewhere)

Other things:
- tcsh vs bash (
- old way of doing conda (PATH modification) vs new way (sourcing their script); the corresponding `conda activate` vs `source activate`

If others have installation tips let me know and we'll try to add them.

### PR DESCRIPTION
**Pull request by larsoner**
Closes #6365.

### CODE REVIEW

**Comment by larsoner** [comment]
@drammock see what you think, feel free to comment here or push directly if you think it could be improved

https://13442-1301584-gh.circle-artifacts.com/0/dev/install/mne_python.html

**Comment by drammock** [comment]
here's the rendering after my tweaks: https://13444-1301584-gh.circle-artifacts.com/0/dev/install/mne_python.html

@larsoner merge if you're happy with them.

**Comment by larsoner** [comment]
That looks better, thanks @drammock

File: doc/install/mne_python.rst [hunk 1]
```diff
@@ -19,64 +21,104 @@ MNE-Python 0.18 only supports Python version 3.5 or higher. We strongly
...
+
+.. collapse:: |hand-stop-o| If you get an error or these look incorrect...
+    :class: danger
+
+    **If you see something like**::
+
+        conda: command not found
+
+    It means that your ``PATH`` variable (what the system uses to find
+    programs) is not set properly. In a correct installation, doing::
```
Comment (jasmainak) [inline doc/install/mne_python.rst]: sorry, late to the party. There is another possibility and something I've observed quite a bit. People install anaconda using one shell and their default is another shell. So while they have it installed, they are just in the wrong shell.

Comment (drammock) [reply doc/install/mne_python.rst]: cool. wanna make a new PR to add that @jasmainak ?

Comment (jasmainak) [reply doc/install/mne_python.rst]: see: https://github.com/mne-tools/mne-python/pull/6404 :)

**Pull request merged**